### PR TITLE
provider/digitalocean: Do not force the creation of a new IP when the droplet_id changes

### LINKED ISSUE #4299: DigitalOcean floating IP destroyed on droplet refresh
This is related to: https://github.com/hashicorp/terraform/issues/2011

Following the example in the docs [0], I'm seeing behavior where terraform destroys a floating IP if the upstream droplet is changed. This seems pretty undesirable since floating IPs are supposed to be static across infrastructure changes, so it seems like the relationship between droplets and floating IPs should be inverted.

[0] https://terraform.io/docs/providers/do/r/floating_ip.html

### PR DESCRIPTION
**Pull request by ColinHebert**
Fixes #4299

The droplet_id isn't mandatory and the expected behaviour of the floating IP is to stick around even when the droplet is destroyed/recreated.

### CODE REVIEW

**Comment by ColinHebert** [comment]
@radeksimko @jen20 Can we go ahead with this PR?

File: builtin/providers/digitalocean/resource_digitalocean_floating_ip.go [hunk 1]
```diff
@@ -73,6 +73,40 @@ func resourceDigitalOceanFloatingIpCreate(d *schema.ResourceData, meta interface
 	return resourceDigitalOceanFloatingIpRead(d, meta)
 }
 
+func resourceDigitalOceanFloatingIpUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*godo.Client)
+
+	if v, ok := d.GetOk("droplet_id"); ok {
```
Comment (jen20) [inline builtin/providers/digitalocean/resource_digitalocean_floating_ip.go]: We should probably check here whether there is a change prior to reassigning the floating IP?

**Comment by jen20** [comment]
Hi @ColinHebert! I've just looked over this - I left a comment inline. Thanks for opening the PR!

**Comment by ColinHebert** [comment]
@jen20 good suggestion. I updated the PR

**Comment by jen20** [comment]
Hi @ColinHebert - looks like there are so `gofmt` issues with this. I'm going to pull it locally and fix it up before merging now.

**Comment by jen20** [comment]
I've fixed up the formatting and squashed these commits into `f6d80f2`, which is now on master. Thanks for you work @ColinHebert!

**Pull request closed without merging**